base/src/main/java/com/smartdevicelink/proxy/SystemCapabilityManager.java

```diff
@@ -415,34 +417,89 @@ public boolean isCapabilitySupported(SystemCapabilityType type){
+		if (callback != null && callback.getSdlMsgVersion() != null) {
+			com.smartdevicelink.util.Version onSystemCapabilityNotificationRPCVersion = new com.smartdevicelink.util.Version(5, 1, 0);
+			com.smartdevicelink.util.Version headUnitRPCVersion = new com.smartdevicelink.util.Version(callback.getSdlMsgVersion().getMajorVersion(), callback.getSdlMsgVersion().getMinorVersion(), callback.getSdlMsgVersion().getPatchVersion());
+			if (headUnitRPCVersion.isNewerThan(onSystemCapabilityNotificationRPCVersion) >= 0) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Checks if the supplied capability type is currently subscribed for or not
+	 * @param systemCapabilityType type of capability desired
+	 * @return true if subscribed and false if not
+	 */
+	private boolean isSubscribedToSystemCapability(SystemCapabilityType systemCapabilityType) {
+		return Boolean.TRUE.equals(systemCapabilitiesSubscriptionStatus.get(systemCapabilityType));
+	}
+
+	/** Gets the capability object that corresponds to the supplied capability type
+	 * <strong>If capability is not cached, the method will return null and trigger the supplied listener when the capability becomes available</strong>
+	 * @param systemCapabilityType type of capability desired
+	 * @param scListener callback to execute upon retrieving capability
+	 * @param subscribe flag to subscribe to updates of the supplied capability type. True means subscribe; false means cancel subscription; null means don't change current subscription status.
+	 * @return desired capability if it is cached in the manager, otherwise returns a null object and works in the background to retrieve the capability for the next call
+	 */
+	public Object getCapability(final SystemCapabilityType systemCapabilityType, final OnSystemCapabilityListener scListener, final Boolean subscribe){
+		Object capability = cachedSystemCapabilities.get(systemCapabilityType);
+		OnSystemCapabilityListener listener = scListener;
+
+		if(capability != null && listener != null) {
+			listener.onCapabilityRetrieved(capability);
+			listener = null; // listener shouldn't be called twice if we end up sending GetSystemCapability request
+		}
+
+		/* GetSystemCapability request should be sent only if:
+		- The capability is not cached
+		- The capability subscription needs to be updated
+		 */
+		boolean shouldUpdateSystemCapabilitySubscription = subscribe != null && subscribe != isSubscribedToSystemCapability(systemCapabilityType) && supportsSubscriptions();
+		if (capability == null || shouldUpdateSystemCapabilitySubscription) {
+			retrieveCapability(systemCapabilityType, listener, subscribe);
+		}
+
+		return capability;
+	}
+
 	/**
 	 * @param systemCapabilityType Type of capability desired
 	 * @param scListener callback to execute upon retrieving capability
 	 */
+	@Deprecated
 	public void getCapability(final SystemCapabilityType systemCapabilityType, final OnSystemCapabilityListener scListener){
```

Comment:
Is it possible to have this simply call `getCapability(type, scListener, subscribe)` above with `subscribe` as `null`?

Reply:
Good idea. I will do that.